Frontend/ui-driver/src/Flow.purs

```diff
@@ -486,6 +487,16 @@ onBoardingFlow = do
     GO_TO_ONBOARD_SUBSCRIPTION -> do
       let onBoardingSubscriptionViewCount =  fromMaybe 0 (fromString (getValueToLocalNativeStore ONBOARDING_SUBSCRIPTION_SCREEN_COUNT))
       onBoardingSubscriptionScreenFlow onBoardingSubscriptionViewCount
+    REFERRAL_CODE_SUBMIT state -> do
+      referDriverResponse <- lift $ lift $ Remote.referDriver (makeReferDriverReq state.data.referralCode)
+      case referDriverResponse of
+        Right (ReferDriverResp resp) -> do
+          modifyScreenState $ RegistrationScreenStateType (\driverReferralScreen -> state{ props{isValidReferralCode = true, referralCodeSubmitted = true, enterReferralCodeModal = false}})
+          setValueToLocalStore REFERRAL_CODE_ADDED "true"
```

Comment:
why local storage is being used?

Reply:
Since once the referral code is applied we can't enter again if the driver kills and reopens the app